File: golang/http2-server.go
package main

// run with (Fish): env PORT=18080 go run http2-server.go
// run with (Bash): PORT=18080 go run http2-server.go

import (
  "fmt"
  "log"
  "net/http"
  "os"
)

func main() {

  // tls_root := "/path/to/tls-certs/"
  tlsRoot := os.Getenv("TESTBED_HOME") + "/etc/tls/"
  port := os.Getenv("PORT")
  if port == "" {
    log.Fatal("Please specify the HTTP port as environment variable, e.g. env PORT=18080 go run http2-server.go")
  }

  srv := &http.Server{Addr: ":" + port, Handler: http.HandlerFunc(handle)}
  log.Fatal(srv.ListenAndServeTLS(tlsRoot + "server.crt", tlsRoot + "server.key"))
}

func handle(w http.ResponseWriter, r *http.Request) {
  // log.Printf("Got connection: %s", r.Proto)
  fmt.Fprintf(w, "Hello World!")
}

File: nodejs/http2-server.js
const http2 = require('http2');
const fs = require('fs');

// const cert_root = '/path/to/certs/';
const cert_root = process.env.TESTBED_HOME + '/etc/tls/';
const server = http2.createSecureServer({
    key: fs.readFileSync(cert_root + 'server.key'),
    cert: fs.readFileSync(cert_root + 'server.crt')
});
server.on('error', (err) => console.error(err));

server.on('stream', (stream, headers) => {
    stream.respond({
        'content-type': 'text/plain; charset=utf-8',
        ':status': 200
    });
    stream.end('Hello world!');
});

server.listen(8080);
